src/Marten/Events/Daemon/HighWater/HighWaterAgent.cs

```diff
@@ -52,15 +52,15 @@ public async Task Start()
         _tracker.Publish(
             new ShardState(ShardState.HighWaterMark, _current.CurrentMark) { Action = ShardAction.Started });
 
-        _loop = Task.Factory.StartNew(DetectChanges, _token,
+        _loop = Task.Factory.StartNew(detectChanges, _token,
             TaskCreationOptions.LongRunning | TaskCreationOptions.AttachedToParent, TaskScheduler.Default);
```

Comment:
FYI this might not be doing what you expect. `StartNew` has no support for async functions, so as soon as the first `await` is hit in `detectChanges`, the thread created from the `LongRunning` flag will be thrown away.

Reply:
I'm shifting that to Task.Run() right now.

Reply:
And using Task.Run() makes the daemon hang, so yes, that was done at least somewhat purposely